### PR TITLE
Add more formula tests for min and max functions

### PR DESCRIPTION
**Pull request by matthias-wende-frequenz**
Also we make the interface public as it's considered ready to use.

### CODE REVIEW

**Comment by shsms** [comment]
In the tests, could you add additional inputs that test both sides of max/min for each case?  Looks like all cases are just testing just one side.

It would be nice to have some `-`, `/` as well,  because those are the trickier parts of formulas.  

Also it would be nice to test cases that have basic operations inside and to the right of min/max.  Like: `a - b.min(c / d) - (c - e)`, etc.

**Review by shsms — DISMISSED** [review]
I did some tests to make sure that this works in strange cases  and I'm now convinced that this works.  So I have no objections to taking this in as it is now.

File: tests/timeseries/_formula_engine/test_formula_composition.py [hunk 1]
```diff
@@ -171,16 +171,70 @@ async def test_formula_composition_missing_bat(self, mocker: MockerFixture) -> N
...
     async def test_formula_composition_min_max(self, mocker: MockerFixture) -> None:
-        """Test the composition of formulas with min/max values."""
+        """Test the composition of formulas with the min and max"""
         mockgrid = MockMicrogrid(grid_meter=True)
+        mockgrid.add_chps(1)
         await mockgrid.start(mocker)
 
         logical_meter = microgrid.logical_meter()
-        engine_min = logical_meter.grid_power._min(  # pylint: disable=protected-access
+        engine_min = logical_meter.grid_power.min(  # pylint: disable=protected-access
```
Comment (daniel-zullo-frequenz) [inline tests/timeseries/_formula_engine/test_formula_composition.py]: do you know why this still needs disable protected-access?

**Review by daniel-zullo-frequenz — COMMENTED** [review]
Only a question/comment to check for and LGTM

File: tests/timeseries/_formula_engine/test_formula_composition.py (hunk 1 above)
Comment (matthias-wende-frequenz) [reply tests/timeseries/_formula_engine/test_formula_composition.py]: I'm pretty sure it's not needed. Good catch!

File: tests/timeseries/_formula_engine/test_formula_composition.py [hunk 2]
```diff
@@ -171,16 +171,70 @@ async def test_formula_composition_missing_bat(self, mocker: MockerFixture) -> N
...
+        mockgrid = MockMicrogrid(grid_meter=True)
+        await mockgrid.start(mocker)
+
+        logical_meter = microgrid.logical_meter()
+        engine_min = logical_meter.grid_power.min(  # pylint: disable=protected-access
             Power.zero()
         ).build("grid_power_min")
         engine_min_rx = engine_min.new_receiver()
-        engine_max = logical_meter.grid_power._max(  # pylint: disable=protected-access
+        engine_max = logical_meter.grid_power.max(  # pylint: disable=protected-access
```
Comment (daniel-zullo-frequenz) [inline tests/timeseries/_formula_engine/test_formula_composition.py]: Then you might want to remove this one too (if it works)
```suggestion
        engine_max = logical_meter.grid_power.max(
```

File: tests/timeseries/_formula_engine/test_formula_composition.py [hunk 3]
```diff
@@ -171,16 +171,70 @@ async def test_formula_composition_missing_bat(self, mocker: MockerFixture) -> N
...
+
+    async def test_formula_composition_min_max_const(
+        self, mocker: MockerFixture
+    ) -> None:
+        """Test the compositing formulas and constants with the min and max functions."""
+        mockgrid = MockMicrogrid(grid_meter=True)
+        await mockgrid.start(mocker)
+
+        logical_meter = microgrid.logical_meter()
+        engine_min = logical_meter.grid_power.min(  # pylint: disable=protected-access
```
Comment (daniel-zullo-frequenz) [inline tests/timeseries/_formula_engine/test_formula_composition.py]:
```suggestion
        engine_min = logical_meter.grid_power.min(
```

**Review by daniel-zullo-frequenz — APPROVED** [review]

**Pull request merged**